### DISCUSSION TITLE
Passing State as Prop and State Updater as Prop

### DISCUSSION THREAD

**Discussion by Android789515** [DioxusLabs/dioxus#3051] [Q&A] [answered]
The docs don't seem to show this in the props section.  We just have simple examples with a button.  How would I pass my signal value down?  The variable is not the same data type as the data it holds.  And how exactly should I pass any updaters down?

**Comment by ealmloff** [comment, marked as answer]
I'm not sure I understand. You can pass Signals in props like normal by adding a field `field: Signal<Data>` to your props. If you read the signal in your child component, it will rerun the component when the signal changes

There is a section in the guide that does this with `ReadOnlySignal` for the `StoryListing` component: https://dioxuslabs.com/learn/0.5/guide/state

**Reply by Android789515** [reply]
That answers my question.  Looks like the signal is essentialy the value and the updater together.  I'd just say the props would be a Signal<Type> or ReadOnlySignal<Type>.
